website/docs/reference/resource-configs/redshift-configs.md

```diff
@@ -257,3 +257,88 @@ The workaround is to execute `DROP MATERIALIZED VIEW my_mv CASCADE` on the data
 </VersionBlock>
 
 </VersionBlock>
+
+<VersionBlock firstVersion="1.8">
+
+## Unit test limitations
+
+Unit tests aren't supported on Redshift if the SQL in the common table expression (CTE) contains functions such as `LISTAGG`, `MEDIAN`, `PERCENTILE_CONT`, etc. Those functions must be executed against a user-created table. dbt combines given rows to be part of the CTE, which is unsupported by Redshift. You can try the following SQL:
+
+```sql
+
+create temporary table "test_tmpxxxxx" as (
+   with test_fixture as (
+       select
+         cast(1000 as integer) as id,
+         cast('menu1' as character varying(500)) as name,
+         cast( 1 as integer) as quantity
+      union all
+      select
+         cast(1001 as integer) as id,
+         cast('menu2' as character varying(500)) as name,
+         cast( 1 as integer) as quantity
+      union all
+      select
+         cast(1003 as integer) as id,
+         cast('menu1' as character varying(500)) as name,
+         cast( 1 as integer) as quantity
+   ),
+   agg as (
+      SELECT
+         LISTAGG(name || ' x ' || quantity, ',') AS option_name_list,
+         id
+      FROM test_fixture
+      GROUP BY id
+   )
+   select * from agg
+);
+
+```
+This results in the error: 
+
+```bash
+
+[XX000] ERROR: One or more of the used functions must be applied on at least one user created tables. Examples of user table only functions are LISTAGG, MEDIAN, PERCENTILE_CONT, etc
+
+```
+
+However, the following query works as expected:
+
+```sql
+
+create temporary table "test_tmp1234" as (
+   SELECT
+      cast(1000 as integer) as id,
+      cast('menu1' as character varying(500)) as name,
+      cast( 1 as integer) as quantity
+   union all
+   select
+      cast(1001 as integer) as id,
+      cast('menu2' as character varying(500)) as name,
+      cast( 1 as integer) as quantity
+   union all
+   select
+      cast(1000 as integer) as id,
+      cast('menu1' as character varying(500)) as name,
+      cast( 1 as integer) as quantity
+);
+
+with agg as (
+   SELECT
+      LISTAGG(name || ' x ' || quantity, ',') AS option_name_list,
+      id
+   FROM test_tmp1234
+   GROUP BY id
+)
+select * from agg;
+
+```
+
+If all given rows are created as a temporary table first, then running the test by referring to it will result in a successful run.
+
+In short, separate the unit tests into two steps:
```

Comment:
I feel like these should be moved up to the top of this section so we start with how to do this right.

Reply:
We do cover these points at the top. Does this work well as a summary, or should we switch the top and the bottom? I think it's beneficial to have it in both places in some fashion

Reply:
I see your point about keeping the summary so instead I [made some suggestions](https://github.com/dbt-labs/docs.getdbt.com/pull/5508/files#r1601956520) to clear up the content at the top.